src/mlpack/methods/ann/augmented/tasks/add_impl.hpp

```diff
@@ -0,0 +1,148 @@
+  arma::field<arma::vec> vecLabels = arma::field<arma::colvec>(batchSize);
+  size_t sizeA = bitLen, sizeB = bitLen;
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    if (!fixedLength)
+    {
+      // Generate random uniform length from [2..maxLength].
+      sizeA = mlpack::math::RandInt(2, bitLen + 1);
+      sizeB = mlpack::math::RandInt(2, bitLen + 1);
+    }
+    // Construct sequence of the form
+    // (binary number with sizeA bits) + '+'
+    // + (binary number with sizeB bits).
+    vecInput(i) = arma::randi<arma::colvec>(
+        sizeA + sizeB + 1, arma::distr_param(0, 1));
+    // Insert special value for '+' delimiter.
+    vecInput(i).at(sizeA) = 0.5;
+
+    int valA = 0;
+    for (size_t k = 0; k < sizeA; ++k)
+    {
+      valA += static_cast<int>(vecInput(i).at(k)) << k;
+    }
+
+    int valB = 0;
+    for (size_t k = sizeA + 1; k < sizeA + 1 + sizeB; ++k)
+    {
+      valB += static_cast<int>(vecInput(i).at(k)) << (k - sizeA - 1);
+    }
+
+    int tot = valA + valB;
+    std::vector<int> binarySeq;
+    while (tot > 0)
+    {
+      binarySeq.push_back(tot & 1);
+      tot >>= 1;
+    }
+    if (binarySeq.empty())
+    {
+      if (valA + valB != 0) {
+        std::ostringstream oss;
+        oss << "AddTask::Generate(): output sequence is empty "
+            << "but the target sum is not 0 (=" << valA + valB << ")"
+            << std::endl;
+        throw std::domain_error(oss.str());
+      }
+      binarySeq.push_back(0);
+    }
+    size_t totLen = binarySeq.size();
+    vecLabels(i) = arma::colvec(totLen);
+    for (size_t j = 0; j < totLen; ++j)
+    {
+      vecLabels(i).at(j) = binarySeq[j];
+    }
+  }
+  Binarize(vecInput, input);
+  Binarize(vecLabels, labels);
+  if (input.n_rows != labels.n_rows) {
+      std::ostringstream oss;
+      oss << "AddTask::Generate(): sequences after application of "
+          << "Binarize() are not aligned ("
+          << input.n_rows << " and " << labels.n_rows << ")"
+          << std::endl;
+      throw std::logic_error(oss.str());
+  }
+  for (size_t i = 0; i < input.n_rows; ++i)
+  {
+    labels.at(i).reshape(input.at(i).n_elem, 1);
+  }
+}
+
+const void AddTask::Generate(arma::mat& input,
+                             arma::mat& labels,
+                             const size_t batchSize)
+{
+  arma::field<arma::mat> fieldInput, fieldLabels;
+  Generate(fieldInput, fieldLabels, batchSize, true);
+  input.set_size(fieldInput(0).n_rows, batchSize);
+  labels.set_size(fieldLabels(0).n_rows, batchSize);
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    input.col(i) = fieldInput.at(i);
+    labels.col(i) = fieldLabels.at(i);
+  }
+}
+
+const void AddTask::Binarize(const arma::field<arma::vec>& input,
+                             arma::field<arma::mat>& output)
+{
+  output = arma::field<arma::mat>(input.n_elem);
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    /*output.at(i) = arma::zeros(3, input.at(i).n_elem);
```

Comment:
This method was needed when we had "one-hot 3d vector" representation to convert stream of numbers (like this `0 0 0 1 2 1 1 0`, where `2` was used as a `+` sign) to vectorized form. Now, since we use that 'stream of bytes' directly as a task representation, we don't really need that.